### PR TITLE
Fix incorrect gas estimate mock and specs

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

There were a few component specs that were incorrectly mocking the `gasFeeEstimates` state as well as asserting the wrong user facing values in the UI. This PR updates those mocks and the specs appropriately.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23081?quickstart=1)

## **Related issues**

Blocking: https://github.com/MetaMask/metamask-extension/pull/23010/files#diff-3e40f7e732d33a982c8929177c57970cd49536a2f49e38e1b11a81b02fe7572c

## **Manual testing steps**

No user facing change as the business logic hasn't changed.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/components/gas-details-item/gas-details-item.test.js [hunk 1]
```diff
@@ -32,7 +32,8 @@ const render = ({ contextProps } = {}) => {
       preferences: {
         useNativeCurrencyAsPrimaryCurrency: true,
       },
-      gasFeeEstimates: mockEstimates[GasEstimateTypes.feeMarket],
+      gasFeeEstimates:
+        mockEstimates[GasEstimateTypes.feeMarket].gasFeeEstimates,
```
Comment (adonesky1) [inline ui/pages/confirmations/components/gas-details-item/gas-details-item.test.js:36]: for further context, the gasFeeEstimates that should have been mocked here were nested one layer deeper than we were previously accessing

File: ui/pages/confirmations/components/confirm-gas-display/confirm-gas-display.test.js [hunk 2]
```diff
@@ -50,7 +51,14 @@ const render = ({ transactionProp = {}, contextProps = {} } = {}) => {
 
 describe('ConfirmGasDisplay', () => {
   it('should match snapshot', async () => {
-    const { container } = render();
```
Comment (adonesky1) [inline ui/pages/confirmations/components/confirm-gas-display/confirm-gas-display.test.js:53]: previously this snapshot was not passed any transaction gasLimit

Comment (jiexi) [reply ui/pages/confirmations/components/confirm-gas-display/confirm-gas-display.test.js:53]: hmm right. maybe we should revert this and update the snapshot again. Do you remember why we did this again? I feel like we were pairing

Comment (jiexi) [reply ui/pages/confirmations/components/confirm-gas-display/confirm-gas-display.test.js:53]: nevermind, understood

**Review by adonesky1 — APPROVED** [review]

**Review by tmashuang — APPROVED** [review]

**Pull request merged**